### PR TITLE
overlord: move install helpers

### PR DESCRIPTION
**Pull request by kubiko**
Move install related helpers from `overlord/devicestate` to the new package `overlord/install`

This will allow the future use of the install helpers from other components (e.g. `snap-bootstrap`) without need to include entire `overlord/devicestate`

### CODE REVIEW

File: overlord/install/export_test.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/secboot"
+	"github.com/snapcore/snapd/secboot/keys"
+	"github.com/snapcore/snapd/seed"
+	"github.com/snapcore/snapd/sysconfig"
+	"github.com/snapcore/snapd/testutil"
+	"github.com/snapcore/snapd/timings"
+)
+
+// // var (
```
Comment (mvo5) [inline overlord/install/export_test.go:38]: This looks like leftovers?

**Review by mvo5 — COMMENTED** [review]
This looks good, tiny comment about what looks like a leftover and it does not pass tests right now, some issues with `export_test.go`.

File: overlord/install/systems.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	{Title: "Factory reset", Mode: "factory-reset"},
+	{Title: "Run normally", Mode: "run"},
+}
+var recoverSystemActions = []SystemAction{
+	{Title: "Reinstall", Mode: "install"},
+	{Title: "Factory reset", Mode: "factory-reset"},
+	{Title: "Run normally", Mode: "run"},
+}
+
+type SeededSystem struct {
```
Comment (pedronis) [inline overlord/install/systems.go:76]: it's a bit unclear to me why we have to move this and related things?

**Comment by pedronis** [comment]
This is quite big and hasn't moved/touched tests and it is moving too much for an initial go at this, in particular it's unclear we can/want to tackle moving bits for the new API and factory reset in a first pass. I'm looking at what is used in the install-only code in the POC and trying to make PRs moving the minimal bits required by that.

**Comment by kubiko** [comment]
Closing this as this has been superseded by other PRs

**Pull request closed without merging**